eth/tracers/firehose.go

```diff
@@ -218,6 +216,16 @@ type Firehose struct {
 	// Testing state, only used in tests and private configs
 	testingBuffer             *bytes.Buffer
 	testingIgnoreGenesisBlock bool
+
+	// Flushing mechanisms
+	flushJobQueue           chan *blockPrintJob
+	flushOrderedOutputQueue chan *blockOutput
+	flushJobWG              sync.WaitGroup
+	flushOrderedOutputWG    sync.WaitGroup
+	flushStarted            bool
+	flushStartBlockNum      uint64
+	flushStartCond          *sync.Cond
+	flushBufferSize         int
```

Comment:
This feels like it grown too much. What about creating a single struct, maybe `BlockFlushQueue` that would wrap all this?